legal-api/src/legal_api/services/business_details_version.py

```diff
@@ -51,22 +63,33 @@ def get_business_revision(transaction_id, business) -> dict:
         business_version = version_class(Business)
         business_revision = db.session.query(business_version) \
             .filter(business_version.transaction_id <= transaction_id) \
+            .filter(business_version.operation_type != 2) \
             .filter(business_version.id == business.id) \
             .filter(or_(business_version.end_transaction_id == None,  # pylint: disable=singleton-comparison # noqa: E711,E501;
                         business_version.end_transaction_id > transaction_id)) \
             .order_by(business_version.transaction_id).one_or_none()
         return VersionedBusinessDetailsService.business_revision_json(business_revision, business.json())
 
     @staticmethod
-    def get_office_revision(transaction_id, business) -> dict:
+    def get_office_revision(transaction_id, business_id) -> dict:
         """Consolidates all office changes upto the given transaction id."""
         offices_json = {}
         address_version = version_class(Address)
+        offices_version = version_class(Office)
 
-        for office in business.offices.all():
+        offices = db.session.query(offices_version) \
+            .filter(offices_version.transaction_id <= transaction_id) \
+            .filter(offices_version.operation_type != 2) \
```

Comment:
What is operation type 2? Should there be an enum for this?

Reply:
Deleted. I dont think an enum is required here